package/network/services/dnsmasq/files/dnsmasq.init

```diff
@@ -380,6 +394,16 @@ dhcp_add() {
 	config_get options "$cfg" options
 	config_get_bool dynamicdhcp "$cfg" dynamicdhcp 1
 
+	if [ $ADD_LOCAL_HOSTNAME -gt 0 ] ; then
+		local dashedname="$( echo $ifname | sed 's/\./-/g' )"
+		local routername="$( uci_get system @system[0] hostname OpenWrt )"
```

Comment:
Hardcoding `OpenWrt` (or `LEDE`) here is not optimal. I suggest to source `/etc/os-release` and to use the `$ID` variable with a fall back to `OpenWrt`.

Example:

    local distribution="$( [ -f /etc/os-release ] && (. /etc/os-release; echo "$ID") || echo openwrt )"
    local routername="$( uci_get system @system[0] hostname "$distribution" )"